plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/DatabricksExecutionContext.java

```diff
@@ -0,0 +1,107 @@
+import org.jkiss.dbeaver.ext.generic.model.GenericSchema;
+import org.jkiss.dbeaver.model.connection.DBPConnectionBootstrap;
+import org.jkiss.dbeaver.model.exec.DBCException;
+import org.jkiss.dbeaver.model.exec.DBCExecutionPurpose;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCStatement;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.utils.CommonUtils;
+
+import java.sql.SQLException;
+
+public class DatabricksExecutionContext extends GenericExecutionContext {
+
+    private static final Log log = Log.getLog(DatabricksExecutionContext.class);
+
+    @Nullable
+    private String activeCatalogName;
+
+    DatabricksExecutionContext(JDBCRemoteInstance instance, String purpose) {
+        super(instance, purpose);
+    }
+
+    @Nullable
+    @Override
+    public GenericCatalog getDefaultCatalog() {
+        if (CommonUtils.isEmpty(activeCatalogName)) {
+            return super.getDefaultCatalog();
+        }
+        return getDataSource().getCatalog(activeCatalogName);
+    }
+
+    @Override
+    public void setDefaultCatalog(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull GenericCatalog catalog,
+        @Nullable GenericSchema schema
+    ) throws DBCException {
+        try {
+            super.setDefaultCatalog(monitor, catalog, schema);
+            activeCatalogName = catalog.getName();
+        } catch (DBCException e) {
+            log.debug("Error setting default catalog for Databricks", e);
+        }
+    }
+
+    @Override
+    public boolean refreshDefaults(DBRProgressMonitor monitor, boolean useBootstrapSettings) throws DBException {
+        boolean isRefreshed = super.refreshDefaults(monitor, useBootstrapSettings);
+
+        try (JDBCSession session = openSession(monitor, DBCExecutionPurpose.META, "Query current catalog")) {
+            try (JDBCStatement dbStat = session.createStatement()) {
+                try (JDBCResultSet dbResult = dbStat.executeQuery("SELECT CURRENT_CATALOG()")) {
+                    if (dbResult != null && dbResult.next()) {
+                        String currentCatalog = dbResult.getString(1);
+
+                        if (CommonUtils.isNotEmpty(currentCatalog)) {
+                            if (!CommonUtils.equalObjects(currentCatalog, activeCatalogName)) {
+                                activeCatalogName = currentCatalog;
+                                isRefreshed = true;
+                            }
+                            if (useBootstrapSettings) {
+                                DBPConnectionBootstrap bootstrap = getBootstrapSettings();
+                                if (CommonUtils.isEmpty(bootstrap.getDefaultCatalogName())) {
+                                    bootstrap.setDefaultCatalogName(currentCatalog);
```

Comment:
You cannot modify bootstrap settings here.
And why?

Reply:
fixed